sdrplay_source/src/main.cpp

```diff
@@ -640,40 +681,61 @@ class SDRPlaySourceModule : public ModuleManager::Instance {
             config.conf["device"] = _this->devNameList[_this->devId];
             config.release(true);
         }
-
-
-        if (ImGui::Combo(CONCAT("##sdrplay_sr", _this->name), &_this->srId, sampleRatesTxt)) {
-            _this->sampleRate = sampleRates[_this->srId];
-            if (_this->bandwidthId == 8) {
-                _this->bandwidth = preferedBandwidth[_this->srId];
-            }
+        
+        ImGui::Text("IF Mode");
+        ImGui::SameLine();
+        ImGui::SetNextItemWidth(menuWidth - ImGui::GetCursorPosX());
+        if (ImGui::Combo(CONCAT("##sdrplay_ifmode", _this->name), &_this->ifModeId, ifModeTxt)) {
+            if(_this->ifModeId != 0){
+                _this->bandwidth = ifModes[_this->ifModeId].bw;
+                _this->sampleRate = ifModes[_this->ifModeId].effectiveSamplerate;
+            } else {
+                config.acquire();
+                _this->sampleRate = config.conf["devices"][_this->selectedName]["sampleRate"];
```

Comment:
This is incorrect, like on load, the correct srID and bwId have to be found (or corrected if the saved samplerate/bandwidth values are invalid).

This would cause a disparity between the samplerate/bandwidth that the UI shows and the one that'll actually be used when starting the device.

Reply:
yeah, i missed that case